juju/juju
# Versioned backup file metadata and cleaner cmd.

## Description of change

Prior to this change, backup files were not versioned.
This PR introduces versions - all previously created files, i.e. all backup files that have no mentioning of version in metadata, are considered to be of version 0.

New version 1 ensures that backup files have controller UUID, controller machine and instance references, as well as the number of HA controller nodes (aka HA configuration) for controllers in HA.

While updating backup CRUD commands to display this changes, I had to re-implement the way metadata output was generated for readability and ease of maintenance.  I have also addressed long-standing question of how these commands deal with --verbose and --quiet options. 

## QA steps

1. bootstrap a controller
2. deploy something
3. use create-backup (both download to the client and stored remotely)
4. enable-ha and wait for it to stabilise
5. use create-backup (again both)
6. use other backup CRUD commands to see output

Note that --verbose and --quiet options change output of backup commands sometimes dramatically.

## Documentation changes

If documentation contains output of backup commands, these may need to be updated.
